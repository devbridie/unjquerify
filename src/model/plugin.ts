import {JQueryExpressionType} from "./matchers/jquery-expression-type";
import {ReturnType} from "./return-types/return-type";
import {Expression, SpreadElement, Statement} from "babel-types";
import {Scope} from "babel-traverse";

export interface Plugin {
    matchesExpressionType: JQueryExpressionType;
    applicableWithArguments: (args: Array<Expression | SpreadElement>) => boolean;
    returnType: ReturnType;
    replaceWith: (element: Expression, args: Expression[], scope: Scope) => Expression | Statement[];
    escapeFromChain?: true;
}
