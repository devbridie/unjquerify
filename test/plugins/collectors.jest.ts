import * as babel from "babel-core";
import {jQueryExpressionPlugin} from "../../src/plugins/jquery-expression.plugin";
import {HidePlugin} from "../../src/plugins/effects/basics/hide.plugin";
import {CssGetPlugin} from "../../src/plugins/manipulation/style-properties/css.get.plugin";
import {QuerySelectorAllPlugin} from "../../src/plugins/selectors/querySelectorAll.plugin";
import {IsPlugin} from "../../src/plugins/traversing/filtering/is.plugin";
import {FindPlugin} from "../../src/plugins/traversing/tree-traversal/find.plugin";

describe("collectors", () => {
    it("chain > value", () => {
        const example = `const size = $("div").hide().css("font-size");`;
        const plugin = jQueryExpressionPlugin([HidePlugin, CssGetPlugin, QuerySelectorAllPlugin]);
        const transformed = babel.transform(example, {plugins: [plugin]}).code;
        expect(transformed).toMatchSnapshot();
    });

    it("chain", () => {
        const example = `const divs = $("div").hide();`;
        const plugin = jQueryExpressionPlugin([HidePlugin, QuerySelectorAllPlugin]);
        const transformed = babel.transform(example, {plugins: [plugin]}).code;
        expect(transformed).toMatchSnapshot();
    });

    it("value", () => {
        const example = `const fontSize = $("div").css("font-size");`;
        const plugin = jQueryExpressionPlugin([CssGetPlugin, QuerySelectorAllPlugin]);
        const transformed = babel.transform(example, {plugins: [plugin]}).code;
        expect(transformed).toMatchSnapshot();
    });

    it("example: is", () => {
        const example = `const matching = $("div").is(".article");`;
        const plugin = jQueryExpressionPlugin([IsPlugin, QuerySelectorAllPlugin]);
        const transformed = babel.transform(example, {plugins: [plugin]}).code;
        expect(transformed).toMatchSnapshot();
    });

    it("example: find", () => {
        const example = `const children = $("div").find(".article");`;
        const plugin = jQueryExpressionPlugin([FindPlugin, QuerySelectorAllPlugin]);
        const transformed = babel.transform(example, {plugins: [plugin]}).code;
        expect(transformed).toMatchSnapshot();
    });
});
