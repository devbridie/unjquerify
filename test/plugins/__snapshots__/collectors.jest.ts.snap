// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collectors chain > value 1`] = `
"const _start = document.querySelectorAll(\\"div\\");

_start.style.display = \\"none\\";
const size = getComputedStyle(_start[0])[\\"font-size\\"];"
`;

exports[`collectors chain 1`] = `
"const divs = document.querySelectorAll(\\"div\\");
divs.forEach(_element => _element.style.display = \\"none\\");"
`;

exports[`collectors example: find 1`] = `"const children = [].concat(...document.querySelectorAll(\\"div\\").map(_element => _element.querySelectorAll(\\".article\\")));"`;

exports[`collectors example: is 1`] = `"const matching = document.querySelectorAll(\\"div\\").some(_element => _element.matches(\\".article\\"));"`;

exports[`collectors value 1`] = `"const fontSize = getComputedStyle(document.querySelectorAll(\\"div\\")[0])[\\"font-size\\"];"`;
